### ISSUE TITLE
Console: Remote schema permissions does not properly handle arguments for non-root fields

### ISSUE THREAD

**Issue by harishnair96** [hasura/graphql-engine#8898]
### Version Information

Server Version: 2.11.1

### Environment

OSS

### What is the current behaviour?

If the non-root field of a remote schema has an argument, setting remote schema permission for that field does not include the argument. This has the effect that roles other than admin role won't be able to use that argument. 

### What is the expected behaviour?

Setting permission on a field must include any arguments that field contains similar to how arguments for root fields are handled.

### How to reproduce the issue?

1. Connect a remote schema to Hasura that has the following GraphQL schema:
  ```graphql
  type Book {
    title: String
    author: String
    category: Category
  }

  type Author {
    name: String
    books(category: Category): [Book]
  }

  enum Category { science, fiction }

  type Query {
    authors: [Author]
  }
  ```
2. Add a permission to that remote schema as shown in the image below:
<img width="1512" alt="add_permission" src="https://user-images.githubusercontent.com/20351766/187845702-a65e4e44-0da9-4bf0-9ae0-c122a3780589.png">

3. Execute the following GraphQL query for the `admin` role -  

  ```graphql
  {
    authors {
      books(category: fiction) {
        title
      }
    }
  }
  ```
4.  Execute the same GraphQL query for the `user` role and observer the error -
  ```json
  {
    "errors": [
      {
        "extensions": {
          "code": "validation-failed",
          "path": "$.selectionSet.authors.selectionSet.books"
        },
        "message": "'books' has no argument named 'category'"
      }
    ]
  }
  ```

### Any possible solutions?

The issue happens because console does not send the proper schema to the backend. Therefore, the permissions for remote schema in the metadata will have the following GraphQL type -
```graphql
type Author {
  books: [Book]
  name: String
}
```
when it should have been -
```graphql
type Author {
  books(category: Category): [Book]
  name: String
}
```
Sending the proper schema to the backend would fix the problem.

### Keywords

remote schema permission

**Comment by harishnair96** [comment]
The workaround for this issue is to edit the metadata manually to use the correct type.

**Comment by rahulagarwal13** [comment]
fixed in 2.13
